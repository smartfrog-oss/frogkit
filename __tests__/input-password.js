import { mount } from 'vue-test-utils'
import InputPassword from '@/components/input-password'

describe('Input password component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(InputPassword, { propsData: { placeholder: 'password' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should toggle input type', () => {
    const wrapper = mount(InputPassword, { propsData: { placeholder: 'password' } })
    expect(wrapper.vm.inputType).toBe('password')
    const icon = wrapper.vm.$el.getElementsByClassName('fk-input-password__img')[0]
    icon.click()
    expect(wrapper.vm.inputType).toBe('text')
    icon.click()
    expect(wrapper.vm.inputType).toBe('password')
  })

})
