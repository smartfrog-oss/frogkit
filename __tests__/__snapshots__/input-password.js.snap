// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input password component should render component and match snapshot 1`] = `
<section class="fk-input-password"><input type="password" value="" block="" placeholder="password" aria-label="password" pid="input_password">
  <div class="fk-input-password__img fk-input-password__show-pass" style="display: none;"></div>
</section>
`;
